src/main/java/ceos/vote/domain/member/service/AuthService.java

```diff
@@ -0,0 +1,197 @@
+import ceos.vote.global.repository.MemberRepository;
+import ceos.vote.global.repository.RefreshRepository;
+import ceos.vote.global.repository.TeamRepository;
+import io.jsonwebtoken.ExpiredJwtException;
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Date;
+
+import static ceos.vote.global.exception.ExceptionCode.*;
+
+@Service
+@Transactional(readOnly = true)
+@RequiredArgsConstructor
+public class AuthService {
+
+    private final MemberRepository memberRepository;
+    private final BCryptPasswordEncoder bCryptPasswordEncoder;
+    private final JWTUtil jwtUtil;
+    private final RefreshRepository refreshRepository;
+    private final TeamRepository teamRepository;
+
+    // [POST] 회원가입
+    @Transactional
+    public MemberResponseDto signup(SignupRequestDto request) {
+
+        String userId = request.userId();
+        String password = request.password();
+
+        if(memberRepository.existsMemberByUserId(userId)){
+            throw new ApplicationException(DUPLICATED_USER_ID);
+        }
+
+        PartType part;
+        TeamType team;
+
+        if (request.part().equals("프론트엔드")) {
+            part = PartType.FRONTEND;
+        } else if (request.part().equals("백엔드")) {
+            part = PartType.BACKEND;
+        } else {
+            throw new ApplicationException(INVALID_PART_TYPE);
+        }
+
+        if (request.team().equals("포토그라운드")) {
+            team = TeamType.PHOTOGROUND;
+        } else if (request.team().equals("엔젤브릿지")) {
+            team = TeamType.ANGELBRIDGE;
+        } else if (request.team().equals("페달지니")) {
+            team = TeamType.PEDALGENIE;
+        } else if (request.team().equals("케이크웨이")) {
+            team = TeamType.CAKEWAY;
+        } else if (request.team().equals("커피딜")) {
+            team = TeamType.COFFEEDEAL;
+        } else {
+            throw new ApplicationException(INVALID_TEAM_TYPE);
+        }
+
+        Team myTeam = teamRepository.findByType(team);
+        Member newMember = request.toEntity(bCryptPasswordEncoder.encode(password), part, myTeam);
+
+        memberRepository.save(newMember);
+
+        return MemberResponseDto.from(newMember);
+    }
+
+    /**
+     * Refresh Token 추출
+     * **/
+    public String extractRefreshToken(HttpServletRequest request) {
+        Cookie[] cookies = request.getCookies();
+        if (cookies != null) {
+            for (Cookie cookie : cookies) {
+                if (cookie.getName().equals("refreshToken")) {
+                    return cookie.getValue();
+                }
+            }
+        }
+        throw new ApplicationException(NOT_FOUND_REFRESH_TOKEN);
+    }
```

Comment:
depth를 줄이는 방법은 if 문을 반대로 생각해보는건데요. `cookies`가 null이면 ApplicationException가 발생하기 때문에 null이 아닌 경우를 처리하기보다는 반대로 cookies가 null인 경우에 예외를 던지도록 변경하면 depth를 하나 줄일 수 있겠죠.
다음으로 cookie에서 꺼내오는 getName에 항상 값이 있다고 보장할 수 있는건가요? 그게 아니라면 세션에서 말씀드린 것과 같이 비교하는 두 요소의 위치를 변경하는 것을 권장드려요. 이렇게 하면 상수를 기준으로 비교하게 되어 NPE 발생을 방지해줘요.

```suggestion
    public String extractRefreshToken(HttpServletRequest request) {
        Cookie[] cookies = request.getCookies();
        if (cookies == null) {
            throw new ApplicationException(NOT_FOUND_REFRESH_TOKEN);
        }
        for (Cookie cookie : cookies) {
           if ("refreshToken".equals(cookie.getName())) {
                return cookie.getValue();
            }
        }
        throw new ApplicationException(NOT_FOUND_REFRESH_TOKEN);
    }
```